### PR TITLE
Fix panic when trace logging is on

### PR DESCRIPTION
**Pull request by lbyte00**

### CODE REVIEW

**Comment by nelsam** [comment]
Have you tried writing tests that expose the issue, so that we don't regress?

**Comment by lbyte00** [comment]
@nelsam how do you check coverage of the code?
`go test -coverprofile=coverage.out` shows that the coverage is 0.0%.

**Comment by nelsam** [comment]
https://github.com/go-gorp/gorp/blob/master/test_all.sh <-- there are coverage statistics set up in here.  Feel free to rely on travis for tests - I know it can suck to set up all of those databases locally.

It's weird that you're getting 0.0% ... while most of `gorp` is tested only with integration tests (it turns out that go's `sql` package makes it pretty hard to unit test), the dialect types all have reasonable tests.  I'll have to dig in to that.

File: gorp_test.go [hunk 1]
```diff
@@ -2360,6 +2360,34 @@ func TestPrepare(t *testing.T) {
...
+func (u UUID4) Value() (driver.Value, error) {
+	if u == "" {
+		return nil, nil
+	}
+
+	return string(u), nil
+}
+
+type NilPointer struct {
+	Id     string
```
Comment (golangci) [inline gorp_test.go]: struct field Id should be ID

File: gorp_test.go [hunk 2]
```diff
@@ -2360,6 +2360,34 @@ func TestPrepare(t *testing.T) {
...
+	UserId *UUID4
```
Comment (golangci) [inline gorp_test.go]: struct field UserId should be UserID

**Comment by nelsam** [comment]
I need to disable GolangCI on this repo :roll_eyes:  Legacy code is legacy.

**Comment by lbyte00** [comment]
@nelsam yeah, looks like so :)

**Comment by lbyte00** [comment]
Coverage is up now 🎉

**Comment by nelsam** [comment]
Looks good to me.  I'm gonna just double-check that the test fails without the fix in place before merge, either tomorrow or this weekend.

**Comment by lbyte00** [comment]
Not sure if you like short variables or not, but this commit will optimize code.

**Comment by golangci** [comment]
Sad to see you disconnected us :(

@nelsam you can disable only `golint` that reports such issues `s/Id/ID`. We have a lot of another useful linters (errcheck, govet, staticcheck, etc) to catch real bugs, not only style issues.
To disable `golint` just add `.golangci.yml` into the root:
```yaml
linters:
  disable:
    - golint
```

You can find documentation about configuration in [README of golangci-lint](https://github.com/golangci/golangci-lint#config-file).

**Comment by nelsam** [comment]
@golangci I agree with all your checks; but there are some things we can't fix because backward compatibility.

I'll probably reintegrate some day, when I get a chance to deprecate and remove the old methods; but it's not worth blocking PRs on those issues today.

**Comment by lbyte00** [comment]
@nelsam hi, are you going to take a look at this issue?

**Comment by nelsam** [comment]
I tried.  Suffice to say, setting up a robot lawn mower took much, much longer than I thought it would.

I should be able to take a look after work today, I think.  Feel free to ping me in gitter chat if you'd like to remind me.

**Comment by nelsam** [comment]
@lbyte00 In case you're curious, I refactored out the else statement after merging (see: a8a0855190d9f7d95b25bd7497868edf048ea3fe).

**Pull request merged**